extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchHolderObjectStrategy.java

```diff
@@ -48,12 +49,18 @@ public int compare(final HllSketchHolder sketch1, final HllSketchHolder sketch2)
   @Override
   public HllSketchHolder fromByteBuffer(final ByteBuffer buf, final int size)
   {
+    if (size == 0 || isSafeToConvertToNullSketch(buf)) {
+      return HllSketchHolder.of((HllSketch) null);
+    }
     return HllSketchHolder.of(HllSketch.wrap(Memory.wrap(buf, ByteOrder.LITTLE_ENDIAN).region(buf.position(), size)));
   }
 
   @Override
   public byte[] toBytes(final HllSketchHolder sketch)
   {
+    if (sketch.getSketch() == null || sketch.getSketch().isEmpty()) {
```

Comment:
Nit: let's rename the variable `sketch` to `holder`.  The code reads funny right now.

Not nit: In the case that this if resolves to false and you return from the final line in this method, the code will have called `sketch.getSketch()` 3 times in this method.  That's 2 more times than it has to and there's no guarantee that `sketch.getSketch()` isn't doing a bunch of work on each independent call.  Yes, you can look at the implementation and see that it's going to be returning a materialized copy on each call after the first, but this code would be significantly improved if it just called `getSketch()` once and then did checks on that return value.  The only time a method should call the exact same getter multiple times in the same execution is when it expects to get different objects on each call.  If it always expects to get the same object, then the code should just get the object once and reuse that reference.  The impact here is minimal, but it is incredibly important that all code be written thinking about how many times methods are called and what references should be getting reused, because there are times when it actually does matter.  It's best to have all code follow good practices always than to pick and choose for this one.

Reply:
Good callout! I think I missed this initially. I'll try to make sure that I stick to this in the future.
Also, renamed the holder parameter.